receiver/prometheusreceiver/metrics_receiver_labels_test.go

```diff
@@ -140,3 +140,102 @@ func TestLabelLimitConfig(t *testing.T) {
 
 	testComponentCustomConfig(t, targets, mp, cfg)
 }
+
+const honorLabelsTarget = `
+# HELP test_gauge0 This is my gauge
+# TYPE test_gauge0 gauge
+test_gauge0{instance="hostname:8080",job="honor_labels_test",testLabel="value1"} 1
+`
+
+func verifyHonorLabelsFalse(t *testing.T, td *testData, rms []*pdata.ResourceMetrics) {
+	want := td.attributes
+	require.Greater(t, len(rms), 0, "At least one resource metric should be present")
+
+	metrics1 := rms[0].InstrumentationLibraryMetrics().At(0).Metrics()
+	ts1 := metrics1.At(0).Gauge().DataPoints().At(0).Timestamp()
+
+	doCompare(t, "honor_labels_false", want, rms[0], []testExpectation{
+		assertMetricPresent("test_gauge0",
+			compareMetricType(pdata.MetricDataTypeGauge),
+			[]dataPointExpectation{
+				{
+					numberPointComparator: []numberPointComparator{
+						compareTimestamp(ts1),
+						compareDoubleValue(1),
+						//job and instance labels must be prefixed with "exported_"
+						compareAttributes(map[string]string{"exported_job": "honor_labels_test", "exported_instance": "hostname:8080", "testLabel": "value1"}),
+					},
+				},
+			}),
+	})
+}
+
+func TestHonorLabelsFalseConfig(t *testing.T) {
+	targets := []*testData{
+		{
+			name: "target1",
+			pages: []mockPrometheusResponse{
+				{code: 200, data: honorLabelsTarget},
+			},
+			validateFunc: verifyHonorLabelsFalse,
+		},
+	}
+
+	mp, cfg, err := setupMockPrometheus(targets...)
+	require.Nilf(t, err, "Failed to create Prometheus config: %v", err)
+
+	testComponentCustomConfig(t, targets, mp, cfg)
+}
+
+func verifyHonorLabelsTrue(t *testing.T, td *testData, rms []*pdata.ResourceMetrics) {
+	//Test for honor_labels: true is skipped. Currently, the Prometheus receiver is unable to support this config
+	//See: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5757
+	t.Skip("skipping test for honor_labels true configuration")
```

Comment:
Skipped tests tend to drift in my experience. I'd prefer to omit the tests, and leave a TODO with a link to the issue to add them.

Reply:
Since we've got an implementation that should work if the appender were able to handle this case I'd like to get this into the history so we can easily resurrect it.  Because we squash PRs to a single commit that means that we'd need to land this and then remove it in a subsequent PR.  I'm fine with leaving it in with a skip but we can follow up to remove this if you think it needs to go.